File: Results/DER/Translation/translation_output/no_test/wizardcoder/codenet/java/python-sanitized/p02336.py
import java.io.IOException;
import java.io.InputStream;
import java.util.*;

public class p02336 implements Runnable {

    public static void main(String[] args) {
        new Thread(null, new p02336(), "", 16 * 1024 * 1024).start();
    }

    public void run() {
        FastScanner sc = new FastScanner();

        int n = sc.nextInt();
        int k = sc.nextInt();

        Combinatrics c = new Combinatrics(n - 1);

        System.out.println(c.homogeneous(k, n - k));

    }
}


class Combinatrics {

File: Results/DER/Translation/translation_output/w_test/granite/codenet/python/java-sanitized/p02336.java
import java.util.*;

public class p02336 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int k = sc.nextInt();
        long ans = 1;
        if (k > n) {
            System.out.println(0);
            return;
        }
        for (int i = 1; i < k; i++) {
            ans *= (n - i);
            ans /= i;
            ans %= 1000000007;
        }
        System.out.println(ans);
    }
}